signal_timeout_block.py

```diff
@@ -67,8 +68,10 @@ def start(self):
                         # scheduled before this persisted job is scheduled, so
                         # only do this if one has not been scheduled already.
                         self._schedule_timeout_job(job, key, interval, True)
+        self._persistence_scheduled.set()
```

Comment:
If I'm understanding this block correctly, we are already stashing away the previous jobs that were persisted in case of some kind of race condition with start. It seems more of the problem here is that we aren't stashing them enough, and are just storing the reference. To me, it makes more sense to just replace that line in `configure` with a `copy(self._repeatable_jobs)`